### ISSUE TITLE
Incorrectly parsing escaped script

### ISSUE THREAD

**Issue by brianjmurrell** [koalaman/shellcheck#2576]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1010, SC1010
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

        ssh root@server set -eu\; data=\"\$\(some_command\)\"\; \
	  if ! echo \"\$data\" \| json_pp\; then \
	    echo \"JSON format could not be pretty-printed.  Raw JSON:\"\; \
	    echo \"\$data\"\; \
	  fi 2>&1

```

#### Here's what shellcheck currently says:
```
Line 4	SC1010: Use semicolon or linefeed before 'then' (or quote to make it literal).
Line 7	SC1010: Use semicolon or linefeed before 'fi' (or quote to make it literal).
```
#### Here's what I wanted or expected to see:

While I can recognize that this is rather complicated with the escaping, I was still hoping shellcheck would be able correctly parse it.

**Comment by koalaman** [comment]
ShellCheck is emitting an warning since unquoted, literal `then` is frequently a bug (a la `if true then echo foo fi`), but does still parse it correctly.

You could choose to ignore the warning (including permanently in `.shellcheckrc` if this is a mistake you're unlikely to make), or quote the `"then"` and `"fi"` to clarify that you did indeed intend these as literal words to be passed to ssh so that the warning is suppressed.

(You could of course also make the whole thing more readable by using one pair of single quotes instead of 20+ backslashes)
